tests/Tests.Shared/test-api.h

```diff
@@ -0,0 +1,599 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+#pragma once
+
+#include "gtest-api.h"
+
+#include "Logger.h"
+#include <mutex>
+#include <string>
+#include <locale>
+#include <codecvt>
+
+#ifdef __OBJC__
+#pragma push_macro("interface")
+#ifndef interface
+#define interface struct
+#endif
+#pragma push_macro("Nil")
+#undef Nil
+#pragma push_macro("__alignof")
+#define __alignof alignof
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wmicrosoft"
+
+#include <UWP/RTHelpers.h>
+#endif
+
+#include <WexTestClass.h>
+
+#ifdef __OBJC__
+#pragma pop_macro("Nil")
+#pragma pop_macro("interface")
+#pragma pop_macro("__alignof")
+#pragma clang diagnostic pop
+#endif
+
+#include "Framework/Framework.h"
+extern void UIApplicationDefaultInitialize();
+
+/////////////////////////////////////////////////////////
+// Logging macros
+/////////////////////////////////////////////////////////
+
+#ifdef __clang__
+#define _TESTAPI_FUNCTION __PRETTY_FUNCTION__
+#else
+#define _TESTAPI_FUNCTION __FUNCTION__
+#endif
+
+#ifdef LOG_INFO
+#undef LOG_INFO
+#endif
+#define LOG_INFO(__format, ...) TestLog::LogComment(GTestLogPrivate::Format(__format, __VA_ARGS__).data())
+
+#ifdef LOG_WARNING
+#undef LOG_WARNING
+#endif
+#define LOG_WARNING(__format, ...) \
+    TestLog::LogWarning(GTestLogPrivate::Format(__format, __VA_ARGS__).data(), __FILE__, _TESTAPI_FUNCTION, __LINE__)
+
+#ifdef LOG_ERROR
+#undef LOG_ERROR
+#endif
+#define LOG_ERROR(__format, ...) \
+    TestLog::LogError(GTestLogPrivate::Format(__format, __VA_ARGS__).data(), __FILE__, _TESTAPI_FUNCTION, __LINE__)
+
+/////////////////////////////////////////////////////////
+// Assert and Verify macros override
+/////////////////////////////////////////////////////////
+
+#ifdef GTEST_SUCCESS_
+#undef GTEST_SUCCESS_
+#endif
+#define GTEST_SUCCESS_(__message) TestLog::LogComment((__message))
+
+#ifdef GTEST_NONFATAL_FAILURE_
+#undef GTEST_NONFATAL_FAILURE_
+#endif
+#define GTEST_NONFATAL_FAILURE_(__message) TestLog::LogError((__message), __FILE__, _TESTAPI_FUNCTION, __LINE__)
+
+#ifdef GTEST_FATAL_FAILURE_
+#undef GTEST_FATAL_FAILURE_
+#endif
+#define GTEST_FATAL_FAILURE_(__message) TestLog::LogErrorAndAbort((__message), __FILE__, _TESTAPI_FUNCTION, __LINE__)
+
+// TODO::
+// todo-nithishm-04182016 - Investigate a better way to support ASSERT_*_MSG and EXPECT_*_MSG macros without re-defining them
+// here. Bug 7244185 is tracking this work.
```

Comment:
Just remove the TODO.  If we didn't need it after so long, we don't *really* need it :)